src/AMSlib/AMS.cpp

```diff
@@ -323,9 +344,92 @@ class AMSWrap
     }
   }
 
+  bool path_exists(std::string &path)
+  {
+    fs::path Path(path);
+    std::error_code ec;
+
+    if (!fs::exists(Path, ec)) {
+      FATAL(AMS, "Path %s does not exist", path.c_str());
+      return false;
+    }
+    return true;
+  }
+
+
+  std::pair<bool, std::string> setup_loggers()
+  {
+    char *ams_logger_level = std::getenv("AMS_LOG_LEVEL");
+    char *ams_logger_dir = std::getenv("AMS_LOG_DIR");
+    char *ams_log_prefix = std::getenv("AMS_LOG_PREFIX");
+    std::string log_path("");
+
+    auto logger = ams::util::Logger::getActiveLogger();
+    bool enable_log = false;
+
+    if (ams_logger_level) {
+      auto log_lvl = ams::util::getVerbosityLevel(ams_logger_level);
+      logger->setLoggingMsgLevel(log_lvl);
+      enable_log = true;
+    }
+
+    if (ams_log_prefix) {
+      // We are going to redirect stdout to some file
+      // By default we store to the current directory
+      std::string log_dir("./");
+      std::string pattern("");
+      std::string log_prefix(ams_log_prefix);
+
+      if (ams_logger_dir) {
+        log_dir = std::string(ams_logger_dir);
+      }
+
+      char hostname[HOST_NAME_MAX];
+      if (gethostname(hostname, HOST_NAME_MAX) != 0) {
+        FATAL(AMS, "Get hostname returns error");
+      }
+
+      CFATAL(AMS, !path_exists(log_dir), "Log Directory does not exist");
```

Comment:
We already have a FATAL in `path_exists`. Also, if the `AMS_LOG_DIR` points to a non-existing directory the code segfault

Reply:
Fixed. We cannot use logs until we setup the loggers. I had to postpone the CFATAL condition.